src/coreclr/jit/ssabuilder.cpp

```diff
@@ -1184,8 +1184,10 @@ void SsaBuilder::BlockRenameVariables(BasicBlock* block)
     {
         if ((memoryKind == GcHeap) && m_pCompiler->byrefStatesMatchGcHeapStates)
         {
+#if !defined(TARGET_WASM)
             // ByrefExposed and GcHeap share any phi this block may have,
             assert(block->bbMemorySsaPhiFunc[memoryKind] == block->bbMemorySsaPhiFunc[ByrefExposed]);
```

Comment:
It means memory liveness on LIR is broken. This is a known upstream issue. Would be nice to add a comment to that effect.